### PR TITLE
Use trivy rootfs scan for both local and remote

### PR DESCRIPTION
**Pull request by partkyle**
This adds support for both the local package and remote registry scanner to use `trivy rootfs` to scan the entire filesystem of a container.

The local version will extract a zarf package and use the images/index.json file to find the available containers, extract each layer into a directory, and then perform `trivy rootfs`

The remote version will write the remote image to a local directory using `github.com/google/go-containerregistry/pkg/v1/layout`, and then modify it to behave exactly the same as the local package so we can use the same process for scanning the remote images. 

We also now only scan for 1 platform: `amd64`. Previously the results of `amd64` and `arm64` were concatenated together, giving double results in most cases.

This resolves the following issues:
- #149 
- #152 
- #153

There is an adr attached for more implementation details.

### CODE REVIEW

File: cmd/store/e2e_test.go [hunk 1]
```diff
@@ -15,17 +15,13 @@ func TestStore(t *testing.T) {
 	const testDBPath = "tests/uds_security_hub.db"
 	github := os.Getenv("GITHUB_TOKEN")
 	ghcrCreds := os.Getenv("GHCR_CREDS")
-	registry1Creds := os.Getenv("REGISTRY1_CREDS")
-	dockerCreds := os.Getenv("DOCKER_IO_CREDS")
-	if github == "" || ghcrCreds == "" || registry1Creds == "" {
-		t.Fatalf("GITHUB_TOKEN, GHCR_CREDS, and REGISTRY1_CREDS are required")
+	if github == "" || ghcrCreds == "" {
```
Comment (naveensrinivasan) [inline cmd/store/e2e_test.go:18]: We would need only one of them. We can take care of this later.

File: docs/adrs/0001-rootfs-scanning.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+# ADR: Scan using `trivy rootfs` to get a complete scan of a package
+
+Date: 2024-08-12
+
+## Status
+
+pending
```
Comment (naveensrinivasan) [inline docs/adrs/0001-rootfs-scanning.md:7]: Accepted

File: pkg/scan/rootfs.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+	if err != nil {
+		return "", fmt.Errorf("failed to write tar file: %w", err)
+	}
+
+	err = os.Mkdir(pkgOutDir, 0o700)
+	if err != nil {
+		return "", fmt.Errorf("failed to create output pkg dir: %w", err)
+	}
+
+	_, _, err = command.ExecuteCommand(
```
Comment (naveensrinivasan) [inline pkg/scan/rootfs.go:36]: Can we please write an issue to make this into a go code? We should do it later. 

I meant `tar` command.

**Review by naveensrinivasan — APPROVED** [review]
Thank you!

**Review by naveensrinivasan — APPROVED** [review]

File: pkg/scan/rootfs.go (hunk 3 above)
Comment (partkyle) [reply pkg/scan/rootfs.go:36]: created: https://github.com/defenseunicorns/uds-security-hub/issues/181

**Pull request merged**